src/dom.ts

```diff
@@ -269,3 +269,67 @@ export function compareElementArea(a: HTMLElement, b: HTMLElement): number {
 
     return aArea - bArea
 }
+
+/** Computes a unique CSS selector for the current HTMLElement.
+ *  Currently buggy, returns a unique selector but doesn't always correspond
+ *  to the selector you should use for functions such as querySelectorAll().
+ */
+export function getSelector(elem: HTMLElement) {
+   if (elem === null || elem === undefined)
+      return undefined
+
+   if (elem == document.body || !elem.parentNode)
+      return "body"
+
+   if (elem.id)
+      return "#" + elem.id
+
+   let parent = elem.parentNode as HTMLElement
+   let n = [].indexOf.call(parent.childNodes, elem) + 1
+   return getSelector(parent) + " :nth-child(" + n + ")"
+}
+
+export const hintworthy_js_elems = {}
+
+/** Adds or removes an element from the hintworthy_js_elems array of the
+ *  current tab.
+ *  This function is made available to the page's. We should be very careful
+ *  about the input it accepts and what it does with it.
+ */
+export function registerEvListenerAction(action: string, ev: string): void {
+   if (!(this instanceof HTMLElement))
+      return
```

Comment:
Is this to avoid hinting e.g. bits of SVGs? Do we actually want to avoid doing that? Sounds like it might be useful.

Reply:
This was to avoid receiving elements such as the window/the whole document. I never thought about SVG elements, replacing this with `!(this instanceof Element)` should work.